modules/openapi-generator/src/test/resources/3_0/oneoOfDiscriminator.yaml

```diff
@@ -0,0 +1,201 @@
+            - type: object
+              properties:
+                fruitType:
+                  type: string
+              required:
+              - fruitType
+      discriminator:
+        propertyName: fruitType
+    FruitReqDisc:
+      oneOf:
+        - $ref: '#/components/schemas/AppleReqDisc'
+        - $ref: '#/components/schemas/BananaReqDisc'
+      discriminator:
+        propertyName: fruitType
+    AppleReqDisc:
+      type: object
+      required:
+        - seeds
+        - fruitType
+      properties:
+        seeds:
+          type: integer
+        fruitType:
+          type: string
+    BananaReqDisc:
+      type: object
+      required:
+        - length
+        - fruitType
+      properties:
+        length:
+          type: integer
+        fruitType:
+          type: string
+    FruitAllOfDisc:
+      oneOf:
+        - $ref: '#/components/schemas/AppleAllOfDisc'
+        - $ref: '#/components/schemas/BananaAllOfDisc'
+      discriminator:
+        propertyName: fruitType
+    AppleAllOfDisc:
+      type: object
+      required:
+        - seeds
+      properties:
+        seeds:
+          type: integer
+      allOf:
+        - $ref: '#/components/schemas/FruitType'
+    BananaAllOfDisc:
+      type: object
+      required:
+        - length
+      properties:
+        length:
+          type: integer
+      allOf:
+        - $ref: '#/components/schemas/FruitType'
+    FruitAnyOfDisc:
+      oneOf:
+        - $ref: '#/components/schemas/AppleAnyOfDisc'
+        - $ref: '#/components/schemas/BananaAnyOfDisc'
+      discriminator:
+        propertyName: fruitType
+    AppleAnyOfDisc:
+      type: object
+      required:
+        - seeds
+      properties:
+        seeds:
+          type: integer
+      anyOf:
+        - $ref: '#/components/schemas/FruitType'
+    BananaAnyOfDisc:
+      type: object
+      required:
+        - length
+      properties:
+        length:
+          type: integer
+      anyOf:
+        - $ref: '#/components/schemas/FruitType'
+    FruitOneOfDisc:
+      oneOf:
+        - $ref: '#/components/schemas/AppleOneOfDisc'
+        - $ref: '#/components/schemas/BananaOneOfDisc'
+      discriminator:
+        propertyName: fruitType
+    AppleOneOfDisc:
+      type: object
+      required:
+        - seeds
+      properties:
+        seeds:
+          type: integer
+      oneOf:
+        - $ref: '#/components/schemas/FruitType'
+    BananaOneOfDisc:
+      type: object
+      required:
+        - length
+      properties:
+        length:
+          type: integer
+      oneOf:
+        - $ref: '#/components/schemas/FruitType'
+    FruitGrandparentDisc:
+      oneOf:
+        - $ref: '#/components/schemas/AppleGrandparentDisc'
+        - $ref: '#/components/schemas/BananaGrandparentDisc'
+      discriminator:
+        propertyName: fruitType
+    Parent:
+      type: object
+      allOf:
+        # the FruitType schema is a grandparent of AppleGrandparentDisc + BananaGrandparentDisc
```

Comment:
`allOf` without discriminator should be composition (Parent _has a_ FruitType, rather than Parent _is a_ FruitType).

Reply:
Definitely, I agree